### PR TITLE
fix for transferring ownership of groupfolders

### PR DESCRIPTION
**Pull request by vanpertsch**
group folder should not be selectable for transferring ownership: https://github.com/nextcloud/groupfolders/issues/1967

Signed-off-by: Vanessa Pertsch <vanessa.pertsch@nextcloud.com>

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

File: apps/files/lib/Controller/TransferOwnershipController.php [hunk 1]
```diff
@@ -97,7 +97,7 @@ public function transfer(string $recipient, string $path): DataResponse {
 			return new DataResponse([], Http::STATUS_BAD_REQUEST);
 		}
 
-		if ($node->getOwner()->getUID() !== $this->userId) {
+		if ($node->getOwner()->getUID() !== $this->userId || !$node->getStorage()->instanceOfStorage(IHomeStorage::class)) {
```
Comment (szaimen) [inline apps/files/lib/Controller/TransferOwnershipController.php:100]: could that potentially introduce problems with external storage or is external storage also an instance of ihomestorage?

Comment (skjnldsv) [reply apps/files/lib/Controller/TransferOwnershipController.php:100]: I'm not really sure transfering an external storage is even supported anyway :thinking: 
@icewind1991 ?

Comment (icewind1991) [reply apps/files/lib/Controller/TransferOwnershipController.php:100]: transferring external storages doesn't make sense imo

Comment (szaimen) [reply apps/files/lib/Controller/TransferOwnershipController.php:100]: Okay

Comment (vanpertsch) [reply apps/files/lib/Controller/TransferOwnershipController.php:100]: @icewind1991 can you approve the pr? (or is there more to be considered?)

File: apps/files/lib/Controller/TransferOwnershipController.php [hunk 2]
```diff
@@ -40,7 +41,6 @@
 use OCP\IRequest;
 use OCP\IUserManager;
 use OCP\Notification\IManager as NotificationManager;
-
 class TransferOwnershipController extends OCSController {
```
Comment (CarlSchwan) [inline apps/files/lib/Controller/TransferOwnershipController.php:44]:
```suggestion

class TransferOwnershipController extends OCSController {
```

**Review by CarlSchwan — APPROVED** [review]

**Pull request merged**